### DISCUSSION TITLE
Setup HTTPS / Encrption - AWS ALB (Load balancer) for https connections to code server?

### DISCUSSION THREAD

**Discussion by paulchill** [coder/code-server#5816] [Q&A] [answered]
**HTTPS / Traffic Encrption**

Can i use an AWS application Load balancer which holds SSL certs and directs traffic (via port 80) to an EC2 where code-server is running, therefor using the SSL certs on the ALB, so it encrypts the code server traffic to the client browser without https warnings?

Or should i need to connect directly to the ec2 and use a self signed certs instead?

So far i have code server set up and i can run with: 
- code-server --bind-addr  x.x.x.x:8080 (private ec2 ip) 
- then go to http://x.x.x.x:8080/ (public ec2 IP) and login to code server without https support

If i change the apache vhost file (ProxyPass etc) that uses the subdomain of the ALB i can:
- go to https:://dev.domain.com and view vscode 
- In the browser it's says https (load balancer cert) but in the code-server console it shows **Not serving HTTPS**

Does this mean that the above is encrypted traffic?

**Comment by jsjoeio** [comment]
Oof...good question. Calling on @code-asher

**Comment by code-asher** [comment, marked as answer]
Connecting through the load balancer and disabling directly connecting to the instance is the right way to do it.  :+1: 

This means that traffic is encrypted from your browser to the load balancer (external to your infrastructure) but not from the load balancer to code-server (internal to your infrastructure).

This is the most common way to set things up so you should be good to go.  Sometimes people will encrypt internal traffic as well but that is usually only because they have to comply with strict security regulations.
